### PR TITLE
fix shellcheck failures in /cluster/addons

### PR DESCRIPTION
**Pull request by mysunshine92**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:

 /kind cleanup

**What this PR does / why we need it**:
fix shellcheck failures in /cluster/addons
./cluster/addons/addon-manager/kube-addons.sh
./cluster/addons/fluentd-elasticsearch/es-image/run.sh
./cluster/addons/fluentd-elasticsearch/fluentd-es-image/run.sh

**Which issue(s) this PR fixes**:

this PR relates to the issue #72956 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by coffeepac — APPROVED** [review]
looks good to me

however, this PR shouldn't mark the related issue as 'FIXED' as there is a lot more work to be done on the upstream issue and github will auto-close the upstream issue once this PR is merged if the comment remains the same. 

I would say 'this issue relates to ...' instead of 'Fixes'

**Comment by mysunshine92** [comment]
> looks good to me

> however, this PR shouldn't mark the related issue as 'FIXED' as there is a lot more work to be done on > the upstream issue and github will auto-close the upstream issue once this PR is merged if the 
> comment remains the same.

> I would say 'this issue relates to ...' instead of 'Fixes'
@coffeepac thank you for your suggestion, i have modified this .

**Comment by mysunshine92** [comment]
cc @coffeepac @eparis

**Comment by mysunshine92** [comment]
cc @coffeepac @eparis @BenTheElder can you help to add  approved, lgtm labels?

**Comment by mysunshine92** [comment]
cc @coffeepac @eparis @BenTheElder can you help to add approved, lgtm labels?

File: cluster/addons/addon-manager/kube-addons.sh
Comment (MrHohn) [inline cluster/addons/addon-manager/kube-addons.sh]: Why exporting this?

File: cluster/addons/addon-manager/kube-addons.sh
Comment (MrHohn) [inline cluster/addons/addon-manager/kube-addons.sh]: Why exporting this?

File: cluster/addons/addon-manager/kube-addons.sh
Comment (MrHohn) [inline cluster/addons/addon-manager/kube-addons.sh]: remove trailing space

File: cluster/addons/addon-manager/kube-addons.sh
Comment (MrHohn) [inline cluster/addons/addon-manager/kube-addons.sh]: Why calling this twice?

**Comment by mysunshine92** [comment]
@MrHohn thank you for your comments, I have modified those.

**Comment by mysunshine92** [comment]
cc @coffeepac @eparis @BenTheElder can you help to add approved, lgtm labels?

File: cluster/addons/addon-manager/kube-addons.sh
Comment (BenTheElder) [inline cluster/addons/addon-manager/kube-addons.sh]: this change looks suspect 🤔

**Comment by spiffxp** [comment]
needs rebase

File: cluster/addons/addon-manager/kube-addons.sh
Comment (spiffxp) [reply cluster/addons/addon-manager/kube-addons.sh]: what shellcheck failure is this addressing?

File: cluster/addons/addon-manager/kube-addons.sh
Comment (spiffxp) [inline cluster/addons/addon-manager/kube-addons.sh]: This is not equivalent behavior, you're now checking the value of the above command's stdout instead of its exit code

What shellcheck failure are you addressing here?

File: cluster/addons/addon-manager/kube-addons.sh
Comment (spiffxp) [inline cluster/addons/addon-manager/kube-addons.sh]: What shellcheck failure are you addressing here?

File: cluster/addons/fluentd-elasticsearch/fluentd-es-image/run.sh
Comment (spiffxp) [inline cluster/addons/fluentd-elasticsearch/fluentd-es-image/run.sh]:
```suggestion
exec /usr/local/bin/fluentd "${FLUENTD_ARGS}"
```

File: cluster/addons/fluentd-elasticsearch/es-image/run.sh
Comment (spiffxp) [inline cluster/addons/fluentd-elasticsearch/es-image/run.sh]: Why is this here? What shellcheck failure are you addressing?

File: cluster/addons/addon-manager/kube-addons.sh
Comment (spiffxp) [inline cluster/addons/addon-manager/kube-addons.sh]: Why `tries[0]`? isn't it a number?

**Review by spiffxp — CHANGES_REQUESTED** [review]
needs some work

**Comment by mysunshine92** [comment]
/assign @eparis
please add an lgtm and approved label, thanks.

**Comment by spiffxp** [comment]
/hold
@mysunshine92 please address the review comments I made above, I'm wary of accepting any of these changes as-is

File: cluster/addons/fluentd-elasticsearch/fluentd-es-image/run.sh
Comment (coffeepac) [reply cluster/addons/fluentd-elasticsearch/fluentd-es-image/run.sh]: Actually, please don't make this change at all, please see:  https://github.com/kubernetes/kubernetes/pull/74187#issuecomment-465870299

We need to split these args.

**Pull request closed without merging**
